src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -512,7 +512,9 @@ static int mixout_process(struct processing_module *mod,
 			continue;
 
 		pending_frames = md->pending_frames[source_index];
-		frames_to_produce = MIN(frames_to_produce, pending_frames);
+
+		if (source->state == COMP_STATE_ACTIVE || pending_frames)
+			frames_to_produce = MIN(frames_to_produce, pending_frames);
```

Comment:
I can understand the !COMP_STATE_ACTIVE case, but how come we can ignore zero pending frames case? Wouldn't this cause audio quality issue if the active source comes back with data on next 1ms tick, but we already mixed out zeros to the output?

Reply:
@kv2019i you mean when that zero-pending source is active? Then we **will** fo the calculation and `frames_to_produce` will be 0

Reply:
Ack @lyakh , so logic is correct.